api/src/main/java/org/apache/apex/api/EmbeddedAppLauncher.java

```diff
@@ -65,6 +66,41 @@ public static EmbeddedAppLauncher newInstance()
   public interface EmbeddedAppHandle extends AppHandle {}
 
   /**
+   * <p>
+   * getDAG.</p>
+   *
+   * @return
+   */
+  public abstract DAG getDAG();
+
+  /**
+   * <p>
+   * cloneDAG.</p>
+   *
+   * @return
+   * @throws java.lang.Exception
+   */
+  public abstract DAG cloneDAG() throws Exception;
```

Comment:
Exception seems to be too broad for the interface/API declaration. Can it be narrowed to a specific checked exception(s). If not, should underlying implementations catch the checked exception and convert it to RuntimeException, if it is necessary.